lib/OpenQA/Worker/Jobs.pm

```diff
@@ -164,6 +175,21 @@ sub upload {
 
     my $res = $OpenQA::Worker::Common::ua->start($tx);
 
+    # Upload known server failures (Instead of anything that's not 200)
+    unless ($res->res->code =~ m/^5/) {
```

Comment:
I am having problems with double negations and such `unless` not in post-condition checks. Would you mind rewriting that with `if`?